### DISCUSSION TITLE
Reaction rate expression

### DISCUSSION THREAD

**Discussion by leilahashemi** [reaktoro/reaktoro#316] [Q&A]
### Description

Hi @allanleal,
May I ask your help with defining reaction rate expression in Reaktoro? I am aiming to follow this paper ([doi:10.1016/j.gca.2010.02.027](https://doi.org/10.1016/j.gca.2010.02.027) ) and use the reaction expression of pyrite reduction to pyrrhotite based on the equation 3, which gives mol/m2 (instead of mol/s). 

$log n_{s(-2)} = log a + b log t + c log PH2 - E_a/(2.303RT)$                                 eq.3

where total sulfide formed (mol/m2), the hydrogen partial pressure (Pascal), the elapsed time (h), and the activation energy (J/mol)

Could you please let me know how can I define such a reaction rate in Reaktoro?

Thank you and kind regards,
Leila

### What is your operating system?

None

### Additional Information

I tried different ideas but none of them worked. Just to keep it short, here are the rate related functions, (I can provide the whole code if it is needed).

```` 
def areafn(props: ChemicalProps): 
    specificSurface = 3.22 #m2/g 
    m = props.speciesAmount("Pyrite") # mole  
    MW = 119.98 # 1 mole of Pyrit = 119.98 gr
    return m*MW*specificSurface  # return current mineral area (in m2)


def ratefn(args: MineralReactionRateModelArgs, dt, i):
    a = 5.98e-6
    b = 0.47
    c = 1.10
    E_a = 52767 #J/mol
    pH2 = 8 #bar
    R = 8.314 #j/mol/K
    specificSurface = 3.22 #m2/g
    amount_Pyrite = 1.5272 #g
    n_S2 = 10**(math.log10(a)+ b * math.log10(dt*i/3600) + c * math.log10(pH2*1e5) - E_a / (2.303 * R * (args.temperature+273.15))) # mol/m2
    area = amount_Pyrite * specificSurface # m2
    return n_S2 * area # mol
    # area = args.area#
    # k = 10**(-2.7569 * (1000/(150 + 273.15))+ 1.0927)  # the reaction rate constant of pyrite at 25 oC (in mol/m2/s)
    # SR = args.Omega
    # return -area * k * (1 - SR)
````

**Comment by allanleal** [comment]
Hi @leilahashemi , The equation in the paper is not in differential form. You would need to derive it with respect to time (which I think is the variable t and not h). This is just based on a quick inspection. Let's have a call to discuss this if you want. I've also noticed that the list of arguments in your tentative rate function has more arguments than expected (it should be just args). The amount of pyrite should change over time and cause the surface area to change accordingly.

**Comment by ehsan-vahabzadeh** [comment]
Hi @allanleal,
I am trying to regenerate and have matched model with paper mentioned in this thread (https://doi.org/10.1016/j.gca.2010.02.027). However, my issue seems to be more basic with the kinetic model.
I want to model the kinetic dissolution of Pyrite and if things works fine, include the Pyrite reduction to Pyrrhotite. When I run the model in equilibrium state, the dissolution and reduction to Pyrrhotite happen, but I could not pull off the kinetic one. The reaction rates are zero. I have tried changing the conditions and kinetic constants, but changed nothing. I was wondering where do you think would be good to look for mistakes?
I have attached the file for both the kinetic and equilibrium ways of calculation below. 

Thanks,
Ehsan
[Pyrite_Reduction_Truche.zip](https://github.com/user-attachments/files/18913886/Pyrite_Reduction_Truche.zip)
